cpp/src/arrow/compute/row/row_internal.cc

```diff
@@ -325,6 +325,7 @@ Status RowTableImpl::AppendSelectionFrom(const RowTableImpl& from,
     // Varying-length rows
     auto from_offsets = reinterpret_cast<const uint32_t*>(from.offsets_->data());
     auto to_offsets = reinterpret_cast<uint32_t*>(offsets_->mutable_data());
+    // TODO: The following two variables are possibly overflowing.
```

Comment:
This is where the "overflow" happens in my other comments. I'll file another issue/PR for it.

Reply:
Did you file the other issue? If so, can you update the TODO here to the proper GH issue number? Thanks!

Reply:
No. I was planning to file it after and refer to this TODO in it.